app/screens/settings/display_clock/display_clock.tsx

```diff
@@ -52,8 +52,8 @@ const DisplayClock = ({componentId, currentUserId, hasMilitaryTimeFormat}: Displ
             value: `${isMilitaryTimeFormat}`,
         };
 
-        savePreference(serverUrl, [timePreference]);
-        close();
+        const updatedSettings = savePreference(serverUrl, [timePreference]);
+        updateSettings(componentId, updatedSettings);
```

Comment:
`updateSettings` will wait for the promise to return before closing. Previously, we were closing right away. Is this change intended/desired?

(Similar in other files)